client/components/fields/editor/EventRelatedPlannings/EmbeddedCoverageForm.tsx

```diff
@@ -81,28 +81,52 @@ function getLanguagesForCoverage(
     };
 }
 
-export class EmbeddedCoverageFormComponent extends React.PureComponent<IProps> {
+interface IState {
+    userList: Array<IUser>;
+    selectedDeskId?: string;
+}
+
+const propertiesToFilterUsersBy = [
```

Comment:
It's not nice that these fields are being duplicated here.

Looking at this again, planning shouldn't be implementing its own custom search. And having user select supporting to limit to desk users would be a useful feature accross superdesk and extensions. Could you instead adjust UserSelect to add `deskId: string` prop? And then use existing `getUserSearchMongoQuery`? It would be like:

```
const query = {
    $and: [
        {_id: {$in: ['id1', 'id2', 'id3']}},
        getUserSearchMongoQuery('abc'),
    ],
};
```